src/include/rbd/librbd.hpp

```diff
@@ -247,6 +247,7 @@ class CEPH_RBD_API Image
   int snap_exists2(const char *snapname, bool *exists);
   int snap_create(const char *snapname);
   int snap_remove(const char *snapname);
+  int snap_remove2(const char *snapname, uint32_t flags);
```

Comment:
It might be nice to add a progress context callback since a flatten will take non-constant time. Same comment for the C++ API.